lib/logflare_web/controllers/plugs/fetch_resource.ex

```diff
@@ -14,22 +14,23 @@ defmodule LogflareWeb.Plugs.FetchResource do
   alias Logflare.Endpoints
   def init(_opts), do: nil
 
-  def call(%{assigns: %{resource_type: type}} = conn, _opts) when is_atom(type) do
-    conn = fetch_query_params(conn)
-
-    resource =
-      case type do
-        :source ->
-          token = Map.get(conn.params, "source")
-          Sources.get_source_by_token(token)
+  def call(%{assigns: %{resource_type: :source}, params: %{"source" => token}} = conn, _opts) do
```

Comment:
I feel that all of this is doing controller-level work. Plugs should be naive from the business domain and this is injecting a lot of business logic into them.